### ISSUE TITLE
Implement optional framerate limiting

### ISSUE THREAD

**Issue by ff-meli** [ff-meli/ImGuiScene#5]
While we currently enforce vsync, there is probably little reason to run at monitor refresh.  Particularly in the opengl implementation this leads to a higher baseline cpu usage for no real gain.

Limiting the entire system to 30-40fps, or separating events from rendering with separate limits, would probably be best for minimizing impact on any hooked applications etc.

This might be as simple as a framerate cap with sleeps in the renderers, but it will need some investigation.

**Comment by ff-meli** [comment]
Also might be nice to significantly reduce framerate, or even pause rendering, when the overlay is not the focused window.  Might need to be user-controllable in case there are dynamic data windows etc.

**Comment by ff-meli** [comment]
Added an optional framerate limiter the SimpleImGuiScene, with the default behavior unchanged (vsync on).  New options allow for unbounded (vsync off, no limit), or an artificial fps cap (vsync off, sleeps for extra time).  These can have a pretty significant impact on cpu usage when the system is otherwise idle, but should reduce it overall in any cases when used.

OpenGL in particular benefits a lot more from a fixed cap, as nvidia's drivers do a Sleep(0) in a loop when waiting for vsync, and basically chew up 20% of your cpu if it's idle (see #9).  A fixed cap (of almost any value at all) will drop that significantly.

I also added an optional flag to pause rendering when the window loses focus, which again has a nice impact on cpu usage when we don't need it.  Dynamic windows that animate or display constantly-updating data will obviously need to leave this flag disabled.

The current recommended usage of everything is to use DX11 where possible, and just leave the default vsync behavior.  If you use OpenGL, set a manual limit of ~50-120 fps.  For both renderers, enable pausing the renderer when it loses focus unless you absolutely need background rendering.
